src/data/shortlist.h

```diff
@@ -210,6 +210,16 @@ class BinaryShortlistGenerator : public ShortlistGenerator {
                            size_t /*trgIdx*/ = 1,
                            bool shared = false);
 
+  // construct directly from buffer
+  BinaryShortlistGenerator(const void* ptr_void,
+                           const size_t blobSize,
+                           Ptr<const Vocab> srcVocab,
+                           Ptr<const Vocab> trgVocab,
+                           size_t srcIdx = 0,
+                           size_t /*trgIdx*/ = 1,
+                           bool shared = false,
+                           bool check = true);
```

Comment:
I saw your check is just `srcVocab == trgVocab`, why take this additionally if you have access to both arguments?

Reply:
Yes, this is a new variable that tells whether the binary file needs to be checked or not.

Reply:
The query remains, but about `shared`, why do you need the extra argument `shared`, if you have access to both `srcVocab `and `trgVocab`, from the arguments to this function and you can infer it within the function?

Reply:
That is a legacy issue. I kept `shared` there for the concern of backward compatibility.

Reply:
Could you leave a note here about the same in comments. Thanks.

Reply:
Sure!